### PR TITLE
Bootstrap v4

### PR DESCRIPTION
**Pull request by metalmatze**
Further improved what @tboerger already started.
I migrated everything from bootstrap v3 to v4. We want to use flexbox and be future prove 😊 

Additionally I integrated gulp to compile our scss to css.
This becomes especially handy when importing bootstrap from within scss to set our own variables before compiling bootstrap's scss to css. We bascially build our own _gitea_ flavor. 🎉 

I don't use webpack because we don't do anything fancy with JS, just scss (css).

This is by far not done at all. It's just a start so we have something to look at already! 😉 

Some screenshots of HD & iPhone 6 Resolutions.

hd
iphone

### CODE REVIEW

**Comment by andreynering** [comment]
I didn't tried locally, but by the pictures looks awesome. LGTM

File: layouts/partials/toplinks.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<li class="nav-item {{ if eq .Site.Params.Toplink " website" }}active{{ end }}">
```
Comment (tboerger) [inline layouts/partials/toplinks.html]: Too much spaces? :)

Comment (tboerger) [reply layouts/partials/toplinks.html]: And indentation missing

File: package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "url": "git+https://github.com/go-gitea/theme.git"
+  },
+  "author": "The Gitea Authors",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/go-gitea/theme/issues"
+  },
+  "homepage": "https://github.com/go-gitea/theme#readme",
+  "dependencies": {
+    "bootstrap": "^4.0.0-alpha.5"
```
Comment (tboerger) [inline package.json:15]: Maybe include font awesome?

**Review by tboerger — COMMENTED** [review]
Does scss still require ruby? Than we should use less

File: package.json (hunk 2 above)
Comment (metalmatze) [reply package.json:15]: Font Awesome is being served via CDN, so I didn't include it in here. 
We actually use bootstrap's source, so I included it. I think we want to use a CDN in the future and not serve it ourselves, right?

File: layouts/partials/toplinks.html (hunk 1 above)
Comment (metalmatze) [reply layouts/partials/toplinks.html]: Where are too much spaces?
Indentation was formatted by my Editor WebStorm. 😊

**Comment by metalmatze** [comment]
We use `gulp-sass` which in turn uses `libsass` which is written in C IIRC.
Nevertheless we can't use less anyway, because bootstrap switched from less to scss from v3 to v4.
This was the right choice of theirs IMHO. :)

File: package.json (hunk 2 above)
Comment (tboerger) [reply package.json:15]: We want to serve the resulting page via a cdn. For the assets I personally prefer to build a bundle.

File: layouts/partials/toplinks.html (hunk 1 above)
Comment (tboerger) [reply layouts/partials/toplinks.html]: A space in front of the if and within the if

**Comment by tboerger** [comment]
Than let's see if drone properly  builds it :)

**Comment by metalmatze** [comment]
It indeed does on the `drone-node` branch. 😊

**Comment by tboerger** [comment]
LGTM

**Pull request merged**